src/messaging/ReliableMessageMgr.cpp

```diff
@@ -520,6 +569,17 @@ void ReliableMessageMgr::CalculateNextRetransTime(RetransTableEntry & entry)
                     backoff.count(), peerIsActive ? "Active" : "Idle", config.mIdleRetransTimeout.count(),
                     config.mActiveRetransTimeout.count(), config.mActiveThresholdTime.count());
 #endif // CHIP_PROGRESS_LOGGING
+
+#if CHIP_CONFIG_MRP_ANALYTICS_ENABLED
+    // For initial send the packet has already been submitted to transport layer successfully.
+    // On re-transmits we do not know if transport layer is unable to retransmit for some
+    // reason. For this reason we only call NotifyMessageSendAnalytics if it is the inital send
+    // and send kRetransmission after we know transmission is successful elsewhere.
+    if (entry.sendCount == 0)
```

Comment:
Why not just put this in `StartRetransmision`?  There you could just comment that this is immediately following the initial attempt to send a message (which, by the way, may _not_ have been "submitted to transport layer successfully"; it might have failed with a non-fatal error that we expect to clear up on retransmit....)

Reply:
Moved to `StartRetransmision` and removed comment